src/backend.rs

```diff
@@ -208,7 +208,14 @@ impl ExtensionImpl<AuthExtension> for AuthBackend {
         request: &AuthRequest,
         resources: &mut ServiceResources<P>,
     ) -> Result<AuthReply> {
-        let fs = &mut resources.filestore(core_ctx);
+        // FIXME: Have a real implementation from trussed
+        let mut backend_path = core_ctx.path.clone();
+        backend_path.push(&PathBuf::from(BACKEND_DIR));
+        let fs = &mut resources.filestore(&CoreContext {
+            path: backend_path,
+            read_dir_state: None,
+            read_dir_files_state: None,
+        });
```

Comment:
Doesn’t this just replace `<client_id>/dat/<backend_id>` with `<backend_id>/dat/<backend_id>`?

Reply:
Yes, but I think this might be what we want?

A backend might want to store keys in its own storage at some point?. Getting rid of it would be much more involved.

Reply:
I'm thinking if we upstream in trussed a mechanism for "backend filestore" it would probably be built  on top of "clientfilestore" like the current "trussedfilestore"

Reply:
I think it should be one of:
1. `<client_id>/<backend_id>/dat`
2. `<client_id>/<backend_id>`

The additional backend directory in `dat` seems unnecessary to me.

Reply:
I removed it. Now debugdumpstore gives:

```
:: PERSISTENT
p"/test\0" p(p"/\0")
p"/test/backend-auth\0" p(p"/test\0")
p"/test/backend-auth/dat\0" p(p"/test/backend-auth\0")
p"/test/backend-auth/dat/pin.00\0" p(p"/test/backend-auth/dat\0")
p"/test/backend-auth/dat/pin.01\0" p(p"/test/backend-auth/dat\0")
p"/trussed\0" p(p"/\0")
p"/trussed/dat\0" p(p"/trussed\0")
p"/trussed/dat/rng-state.bin\0" p(p"/trussed/dat\0")
:: VOLATILE
```